#### Assignment 3_

#### Statement of Problem:

In [11]:
# Using the 'Boston' dataset provided, please answer the following questions;

# showing clearly the methodology employed to arrive at each answer:

In [2]:
# importing the necessary libraries 
import pandas as pd;
import matplotlib.pyplot as plt;
import numpy as np;

In [3]:
# importing the csv file onto our file:
boston_data = pd.read_csv(r'/home/jeromemugita/Downloads/Boston - Boston.csv')

In [4]:
df = boston_data.copy()

In [ ]:
# 1. What type of file is this dataset ?

##### the file we are using is a .csv (comma separated values) file

In [6]:
# 2. How many rows and columns does it consist of ?

df.shape

(506, 15)

##### here we see that the .shape attribute returns a data model with 506 rows and 15 colums

In [10]:
# 3.Calculate the Interquartile Range 

interQR = np.quantile(df, 0.75) - np.quantile(df, 0.25)
interQR

43.8

##### here we find that the inter-quantile range for our dataset is 43.8

In [13]:
# 4. What is the Standard deviation ?

np.sqrt(np.var(df, ddof=1))

Unnamed: 0    146.213884
crim            8.601545
zn             23.322453
indus           6.860353
chas            0.253994
nox             0.115878
rm              0.702617
age            28.148861
dis             2.105710
rad             8.707259
tax           168.537116
ptratio         2.164946
black          91.294864
lstat           7.141062
medv            9.197104
dtype: float64

##### we can see that the results show the standard deviations for each of our dataset's columns

In [18]:
# 5. What is the mean,median and mode values ?
df.agg([np.mean, np.median, max])

Unnamed: 0       crim          zn      indus     chas       nox   
mean         253.5   3.613524   11.363636  11.136779  0.06917  0.554695  \
median       253.5   0.256510    0.000000   9.690000  0.00000  0.538000   
max          506.0  88.976200  100.000000  27.740000  1.00000  0.871000   

              rm         age        dis        rad         tax    ptratio   
mean    6.284634   68.574901   3.795043   9.549407  408.237154  18.455534  \
median  6.208500   77.500000   3.207450   5.000000  330.000000  19.050000   
max     8.780000  100.000000  12.126500  24.000000  711.000000  22.000000   

             black      lstat       medv  
mean    356.674032  12.653063  22.532806  
median  391.440000  11.360000  21.200000  
max     396.900000  37.970000  50.000000

##### the solution above contains data representing the mean, median and the mode (max - most responses for each column). 

In [15]:
# 6. List the various data types in the data set.

df.dtypes

Unnamed: 0      int64
crim          float64
zn            float64
indus         float64
chas            int64
nox           float64
rm            float64
age           float64
dis           float64
rad             int64
tax             int64
ptratio       float64
black         float64
lstat         float64
medv          float64
dtype: object

##### the result above shows the different datatypes for our different column entries. 